app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -56,6 +102,25 @@ const SurveyAnimals: React.FC = () => {
     }
   };
 
+  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDevice) => {
+    const critter = critterData?.find((a) => a.survey_critter_id === survey_critter_id);
+    const critterTelemetryDevice = { ...data, critter_id: critter?.critter_id ?? '' };
+    try {
+      await bhApi.survey.addDeployment(projectId, surveyId, survey_critter_id, critterTelemetryDevice);
+    } catch (e) {
+      dialogContext.setSnackbar({
+        open: true,
+        snackbarMessage: (
+          <Typography variant="body2" component="div">
+            {`Could not add deployment.`}
+          </Typography>
+        )
+      });
+    }
+    setOpenAddDeviceDialog(false);
```

Comment:
Should add this to a `finally {}` branch if we want it to run after either the `try` or `catch`